pkg/api/componentreadiness/test_details.go

```diff
@@ -25,55 +25,195 @@ import (
+			}
+			testKeyTestJobRunStatuses[testKeyStr].BaseOverrideStatus[jobName] =
+				append(testKeyTestJobRunStatuses[testKeyStr].BaseOverrideStatus[jobName], row)
+		}
+	}
+	for jobName, rows := range allTestsJobRunStatuses.SampleStatus {
+		for _, row := range rows {
+			testKeyStr := row.TestKeyStr
+			if _, ok := testKeyTestJobRunStatuses[testKeyStr]; !ok {
+				testKeyTestJobRunStatuses[testKeyStr] = crtype.TestJobRunStatuses{
+					BaseStatus:         map[string][]crtype.TestJobRunRows{},
+					BaseOverrideStatus: map[string][]crtype.TestJobRunRows{},
+					SampleStatus:       map[string][]crtype.TestJobRunRows{},
+					GeneratedAt:        allTestsJobRunStatuses.GeneratedAt,
+				}
+			}
+			if testKeyTestJobRunStatuses[testKeyStr].SampleStatus[jobName] == nil {
+				testKeyTestJobRunStatuses[testKeyStr].SampleStatus[jobName] = []crtype.TestJobRunRows{}
+			}
+			testKeyTestJobRunStatuses[testKeyStr].SampleStatus[jobName] =
+				append(testKeyTestJobRunStatuses[testKeyStr].SampleStatus[jobName], row)
+		}
+	}
+
+	reports := []crtype.ReportTestDetails{}
+	for _, tOpt := range c.ReqOptions.TestIDOptions {
+		testKey := crtype.TestWithVariantsKey{
+			TestID:   tOpt.TestID,
+			Variants: tOpt.RequestedVariants,
+		}
+		testKeyStr := testKey.KeyOrDie()
+		if statuses, ok := testKeyTestJobRunStatuses[testKeyStr]; ok {
+			report, errs2 := c.GenerateDetailsReportForTest(ctx, tOpt, statuses)
+			if len(errs2) > 0 {
+				errs = append(errs, errs2...)
+				continue
+			}
+			reports = append(reports, report)
+		} else {
+			logrus.Errorf("missing test key in results: %v", testKeyStr)
+
+		}
+
+	}
+	return reports, errs
+}
+
+// GenerateDetailsReportForTest generates a test detail report for a per-test + variant combo.
+func (c *ComponentReportGenerator) GenerateDetailsReportForTest(ctx context.Context, testIDOption crtype.RequestTestIdentificationOptions, componentJobRunTestReportStatus crtype.TestJobRunStatuses) (crtype.ReportTestDetails, []error) {
+
+	if testIDOption.TestID == "" {
+		return crtype.ReportTestDetails{}, []error{fmt.Errorf("test_id has to be defined for test details")}
+	}
+	for _, v := range c.ReqOptions.VariantOption.DBGroupBy.List() {
+		if _, ok := testIDOption.RequestedVariants[v]; !ok {
+			return crtype.ReportTestDetails{}, []error{
+				fmt.Errorf("all dbGroupBy variants have to be defined for test details: %s is missing in %v",
+					v, testIDOption.RequestedVariants),
+			}
+		}
+	}
+
+	releases, errs := query.GetReleaseDatesFromBigQuery(ctx, c.client, c.ReqOptions)
+	if errs != nil {
+		return crtype.ReportTestDetails{}, errs
+	}
+
 	now := time.Now()
 	componentJobRunTestReportStatus.GeneratedAt = &now
 
+	// TODO: this is the spot, here we would have base and sample status, for all MultiTestIDOptions
+	// sort them by test and variant
+	// invoke report for each
+	// refactor so we can get multiple reports from one query, break out everything below.
```

Comment:
was this meant to be left here?

Reply:
My bad, that was where I originally thought it had to happen.